### ISSUE TITLE
RCS1213 false positive

### ISSUE THREAD

**Issue by wmjordan** [JosefPihrt/Roslynator#558]
**Product and Version Used**: 
VS 2019 16.1, latest Roslynator

**Steps to Reproduce**:
Create a new project, with the following code file:
````CSharp
using System;
using System.Collections.Generic;

namespace X
{
	static class Class1
	{
		static readonly Dictionary<string, Do> Methods = new Dictionary<string, Do> {
			{ "a", P }
		};
		delegate string Do(int n);

		static string P(this int n) { return null; }

		public static void Work () {
			Methods["a"](1);
		}
	}
}
````
**Actual Behavior**:
Delegate `Do` was marked unused.

**Expected Behavior**:
Delegate `Do` should not be marked unused.
